### ISSUE TITLE
Error while running rvm mount

### ISSUE THREAD

**Issue by thbar** [oracle/truffleruby#1070]
After my first foray in #1054, I'm trying to setup the latest version of TruffleRuby via RVM, in order to be able to compare behaviours more easily with MRI & write reproductions for problems I've met etc.
 
I'm on Mac, and unpacked the GraalVM tgz under `~/graalvm/graalvm-0.30.2`.

Then I tried mounting with RVM (1.29.3), but got errors:

```
$ rvm mount graalvm/graalvm-0.30.2/Contents/Home/jre/bin/ruby -n truffleruby-0.30.2
Mounting 'ext-truffleruby-0.30.2' from '/Users/thbar/graalvm/graalvm-0.30.2/Contents/Home/jre/languages/ruby' with 'bin/ruby'
ext-truffleruby-0.30.2 - #gemset created /Users/thbar/.rvm/gems/ext-truffleruby-0.30.2@global
ext-truffleruby-0.30.2 - #importing gemset /Users/thbar/.rvm/gemsets/global.gems..................................-
ext-truffleruby-0.30.2 - #generating global wrappers.........
Error running 'run_gem_wrappers_regenerate',
please read /Users/thbar/.rvm/log/1517685212_ext-truffleruby-0.30.2/gemset.wrappers.global.log
ext-truffleruby-0.30.2 - #gemset created /Users/thbar/.rvm/gems/ext-truffleruby-0.30.2
ext-truffleruby-0.30.2 - #importing gemsetfile /Users/thbar/.rvm/gemsets/default.gems evaluated to empty gem list
ext-truffleruby-0.30.2 - #generating default wrappers.........
Error running 'run_gem_wrappers_regenerate',
please read /Users/thbar/.rvm/log/1517685268_ext-truffleruby-0.30.2/gemset.wrappers.default.log
```

I think it tried to generate the wrappers, it failed, then it retried, then it failed again.

I've saved the first log here: 

https://gist.github.com/thbar/82d790be1cba95e28dda960e83522be8

(the second log seems to contain almost exactly the same output).

After that I can actually use ruby, but I'll get error (actually more like warnings) when invoking certain commands:

```
$ rvm ext-truffleruby-0.30.2
Using /Users/thbar/.rvm/gems/ext-truffleruby-0.30.2
$ ruby -v
truffleruby 0.30.2, like ruby 2.3.5 <Java HotSpot(TM) 64-Bit Server VM 1.8.0_151-b12 with Graal> [darwin-x86_64]
$ which gem
/Users/thbar/.rvm/rubies/ext-truffleruby-0.30.2/bin/gem
$ gem --version
[ruby] WARN A nonstandard GEM_HOME is set /Users/thbar/.rvm/gems/ext-truffleruby-0.30.2
[ruby] WARN The bad GEM_HOME may come from a ruby manager, make sure you've called one of: `rvm use system`, `rbenv system`, or `chruby system` to clear the environment.
Error loading RubyGems plugin "/Users/thbar/.rvm/gems/ext-truffleruby-0.30.2@global/gems/executable-hooks-1.3.2/lib/rubygems_plugin.rb": cannot load such file -- executable-hooks/wrapper (LoadError)
Error loading RubyGems plugin "/Users/thbar/.rvm/gems/ext-truffleruby-0.30.2@global/gems/gem-wrappers-1.3.2/lib/rubygems_plugin.rb": cannot load such file -- gem-wrappers (LoadError)
2.5.2.1
```

(the version is displayed ultimately).

**Comment by thbar** [comment]
If I follow the hint indicated [here](https://github.com/oracle/truffleruby/blob/master/doc/user/ruby-managers.md#using-truffleruby-without-a-ruby-manager):

```
$ unset GEM_HOME GEM_PATH GEM_ROOT
```

then the warnings disappear, but not the error:

```
$ gem --version
Error loading RubyGems plugin "/Users/thbar/graalvm/graalvm-0.30.2/Contents/Home/jre/languages/ruby/lib/ruby/gems/2.3.0/gems/executable-hooks-1.3.2/lib/rubygems_plugin.rb": cannot load such file -- executable-hooks/wrapper (LoadError)
Error loading RubyGems plugin "/Users/thbar/graalvm/graalvm-0.30.2/Contents/Home/jre/languages/ruby/lib/ruby/gems/2.3.0/gems/gem-wrappers-1.3.2/lib/rubygems_plugin.rb": cannot load such file -- gem-wrappers (LoadError)
2.5.2.1
```

**Comment by ricsdeol** [comment]
Hi, it's happened with me.
I'm installed  GraalVM Community Edition 1.0 RC1and add component by `gu -c install org.graalvm.ruby`

and add config rvm (https://github.com/oracle/truffleruby/blob/master/doc/user/ruby-managers.md#rvm)

but don't work: 

Errors:

> rvm mount  ~/.opt/graalvm/jre/languages/ruby -n truffleruby
Mounting 'ext-truffleruby' from '/home/ricsdeol/.opt/graalvm-1.0.0-rc1/jre/languages/ruby' with 'bin/ruby'
ext-truffleruby - #gemset created /home/ricsdeol/.rvm/gems/ext-truffleruby@global
ext-truffleruby - #importing gemset /home/ricsdeol/.rvm/gemsets/global.gemsthere was an error installing gem gem-wrappers
there was an error installing gem rubygems-bundler
.....................................................-there was an error installing gem rvm
there was an error installing gem bundler
there was an error installing gem rubocop
....................................................../there was an error installing gem pry-byebug
there was an error installing gem httparty
.........................
ext-truffleruby - #generating global wrappers..................
Error running 'run_gem_wrappers_regenerate',
please read /home/ricsdeol/.rvm/log/1524256402_ext-truffleruby/gemset.wrappers.global.log
ext-truffleruby - #gemset created /home/ricsdeol/.rvm/gems/ext-truffleruby
ext-truffleruby - #importing gemsetfile /home/ricsdeol/.rvm/gemsets/default.gems evaluated to empty gem list
ext-truffleruby - #generating default wrappers..................
Error running 'run_gem_wrappers_regenerate',
please read /home/ricsdeol/.rvm/log/1524256422_ext-truffleruby/gemset.wrappers.default.log
╭─ricsdeol at ricsdeol-solvian in ~/bin using ‹ruby-2.4.1› 18-04-20 - 17:33:45
╰─⠠⠵ rvm use ext-truffleruby
Using /home/ricsdeol/.rvm/gems/ext-truffleruby
╭─ricsdeol at ricsdeol-solvian in ~/bin using ‹ext-truffleruby› 18-04-20 - 17:33:49
╰─⠠⠵ ruby --version
truffleruby 1.0.0-rc1, like ruby 2.3.7 <native build with Graal> [x86_64-linux]
╭─ricsdeol at ricsdeol-solvian in ~/bin using ‹ext-truffleruby› 18-04-20 - 17:33:54
╰─⠠⠵ gem install bundler
[ruby] WARN A nonstandard GEM_HOME is set /home/ricsdeol/.rvm/gems/ext-truffleruby
[ruby] WARN The bad GEM_HOME may come from a ruby manager, make sure you've called one of: `rvm use system`, `rbenv system`, or `chruby system` to clear the environment.
ERROR:  Loading command: install (RuntimeError)
	java.io.IOException: Error while trying to parse null (RuntimeException)
	from java.lang.Throwable.<init>(Throwable.java:310)
	from java.lang.Exception.<init>(Exception.java:102)
	from java.lang.RuntimeException.<init>(RuntimeException.java:96)
	from com.oracle.truffle.api.TruffleLanguage.parse(TruffleLanguage.java:1062)
	from com.oracle.truffle.api.TruffleLanguage$Env.parse(TruffleLanguage.java:1568)
	from org.truffleruby.language.loader.FeatureLoader.lambda$loadCExtLibrary$2(FeatureLoader.java:279)
	from org.truffleruby.language.loader.FeatureLoader$$Lambda$1224/1911388477.accept(Unknown Source)
	from org.truffleruby.cext.Linker.loadLibrary(Linker.java:135)
	from org.truffleruby.language.loader.FeatureLoader.loadCExtLibrary(FeatureLoader.java:275)
	from org.truffleruby.language.loader.RequireNode.requireCExtension(RequireNode.java:172)
Caused by:
Error while trying to parse null (IOException)
	from java.lang.Throwable.<init>(Throwable.java:287)
	from java.lang.Exception.<init>(Exception.java:84)
	from java.io.IOException.<init>(IOException.java:81)
	from com.oracle.truffle.llvm.Runner.parse(Runner.java:273)
	from com.oracle.truffle.llvm.Sulong.parse(Sulong.java:114)
	from com.oracle.truffle.api.TruffleLanguage$ParsingRequest.parse(TruffleLanguage.java:596)
	from com.oracle.truffle.api.TruffleLanguage.parse(TruffleLanguage.java:1058)
	from com.oracle.truffle.api.TruffleLanguage$Env.parse(TruffleLanguage.java:1568)
	from org.truffleruby.language.loader.FeatureLoader.lambda$loadCExtLibrary$2(FeatureLoader.java:279)
	from org.truffleruby.language.loader.FeatureLoader$$Lambda$1224/1911388477.accept(Unknown Source)
Caused by:
External function @SSLv23_method cannot be found. (LinkageError)
	from java.lang.Throwable.<init>(Throwable.java:265)
	from java.lang.Error.<init>(Error.java:70)
	from java.lang.LinkageError.<init>(LinkageError.java:55)
	from com.oracle.truffle.llvm.runtime.NFIContextExtension.getNativeFunction(NFIContextExtension.java:293)
	from com.oracle.truffle.llvm.runtime.LLVMFunctionDescriptor$UnresolvedFunction.resolve(LLVMFunctionDescriptor.java:213)
	from com.oracle.truffle.llvm.runtime.LLVMFunctionDescriptor$UnresolvedFunction.createNativeWrapper(LLVMFunctionDescriptor.java:231)
	from com.oracle.truffle.llvm.runtime.LLVMFunctionDescriptor.toNative(LLVMFunctionDescriptor.java:495)
	from com.oracle.truffle.llvm.runtime.LLVMFunctionDescriptor$LLVMFunctionDescriptorNativeLibrary.toNative(LLVMFunctionDescriptor.java:458)
	from com.oracle.truffle.llvm.runtime.LLVMFunctionDescriptor$LLVMFunctionDescriptorNativeLibrary.toNative(LLVMFunctionDescriptor.java:436)
	from com.oracle.truffle.llvm.runtime.nodes.api.LLVMObjectNativeFactory$CachedToNativeNode.toNative(LLVMObjectNativeFactory.java:232)
Translated to internal error
ERROR:  While executing gem ... (NoMethodError)
    undefined method `invoke_with_build_args' for nil:NilClass

**Comment by Confusion** [comment]
BTW, I think the GEM_HOME warnings are wrong: that path seems like the correct one when using truffleruby via rvm? In my environment (see below) I'm seeing those warnings, but not that error. Many gems install fine.

*Environment*
Ubuntu 16.04
rvm 1.29.3
GraalVM/TruffleRuby 1.0.0-rc1

**Comment by eregon** [comment]
Right, the warning should not be triggered with `~/.rvm/gems/ext-truffleruby`.
@pitr-ch Could you take a look at that?

**Comment by eregon** [comment]
This was fixed by @pitr-ch in [1.0.0-rc2](https://github.com/oracle/truffleruby/blob/master/CHANGELOG.md#10-rc-2-may-2018):
> Imprecise wrong gem directory detection was replaced. TruffleRuby newly marks its gem directories with a marker file, and warns if you try to use TruffleRuby with a gem directory which is lacking the marker.

`rvm mount` as above + `gem --version` works fine now, with no warnings.

However, there is one tricky bit: the post-install hook *must* be run before running `rvm mount` (otherwise all `gem` invocations might fail as the openssl extension might be compiled against a different libssl version).
This is already mentioned when installing the ruby component:
```
$ bin/gu install ruby
Downloading: Component catalog
Processing component archive: Component ruby
Downloading: Component ruby
Installing new component: TruffleRuby (org.graalvm.ruby, version 1.0.0-rc3)

IMPORTANT NOTE:
---------------
The Ruby openssl C extension needs to be recompiled on your system to work with the installed libssl.
Make sure headers for libssl are installed, see https://github.com/oracle/truffleruby/blob/master/doc/user/installing-libssl.md for details.
Then run the following command:
        /home/eregon/code/graalvm-ce-1.0.0-rc3/jre/languages/ruby/lib/truffle/post_install_hook.sh
```

But it would be good to mention it again in the [documentation about `rvm mount`](https://github.com/oracle/truffleruby/blob/master/doc/user/ruby-managers.md#rvm-1).

Also, the existing message should be updated mention about LLVM as that's needed as well.

**Comment by eregon** [comment]
I updated the documentation in cdd6439dc4f56fdd4529e8171fab66503a58860c, it should be clear now.
